nemo_aligner/utils/distributed.py

```diff
@@ -23,10 +23,13 @@
 from typing import Optional
 
 import torch
-from megatron.core import tensor_parallel
+from megatron.core import parallel_state, tensor_parallel
+from megatron.core.models.gpt import GPTModel as MCoreGPTModel
+from megatron.core.pipeline_parallel.schedules import get_forward_backward_func
 
+from nemo.collections.nlp.modules.common.megatron.utils import get_iterator_k_split
+from nemo.collections.nlp.parts import utils_funcs
 from nemo.utils.timers import NamedTimer
-from nemo_aligner.utils import parallel_state
```

Comment:
@ashors1 @gshennvm 
What is the difference between `megatron.core.parallel_state` and `nemo_aligner.utils.parallel_state`? 
Since the `main` is using `nemo_aligner.utils.parallel_state`, I guess we should use it as well. @gshennvm Can you double check?

Reply:
@gshennvm any idea?

Reply:
we should use aligner's parallel_state. the reason is because we need to do monkey patching of the parallelism groups to support resharding in RLHF. it's an opt in feature but best to keep it consistent